projects/sonar/src/app/record/document/document.component.ts

```diff
@@ -115,46 +84,16 @@ export class DocumentComponent implements ResultItem, OnDestroy, OnInit {
   }
 
   /**
-   * Load the thumbnail for record
+   * Return the main file of the record.
    */
-  private _loadThumbnail() {
-    if (!this.record.metadata._files) {
-      return;
-    }
-
-    // Try to find the main file. This is the first file, because files are
-    // already ordered during REST API call.
-    const mainFile = this.record.metadata._files.find(
-      (file: any) => file.type === 'file'
-    );
-
-    if (!mainFile) {
-      return;
-    }
-
-    // Find filename without extension.
-    const matches = mainFile.key.match(/^(.*)\..*$/);
-
-    if (!matches) {
-      return;
-    }
-
-    // Find thumbnail corresponding to filename.
-    const thumbnail = this.record.metadata._files.find(
-      (file: any) =>
-        file.type === 'thumbnail' && file.key === `${matches[1]}.jpg`
-    );
-
-    if (!thumbnail) {
-      return;
+  get mainFile(): DocumentFile {
+    if (!this.record.metadata._files || this.record.metadata._files.length === 0) {
+      return null;
     }
 
-    if (this.restricted === true) {
-      this.thumbnail = '/static/images/restricted.png';
-      return;
-    }
+    const files = this.record.metadata._files.filter((file: any) => file.type === 'file');
 
-    this.thumbnail = `/documents/${this.record.metadata.pid}/files/${thumbnail.key}`;
+    return files.length === 0 ? null : files[0];
```

Comment:
Is that means that you do not have the notion of main File during the deposit procedure?

Reply:
As discussed, the main file is always a the file with `order` set to `0`. Maybe in the future, it will be more convenient to store this information in a dedicated property.